platform/ios/app/MBXViewController.m

```diff
@@ -1530,6 +1542,51 @@ - (void)testQueryPointAnnotations {
     [self presentViewController:alertController animated:YES completion:nil];
 }
 
+- (id<MGLAnnotation>)randomOffscreenAnnotation {
+    NSArray *annotations = self.mapView.annotations;
+
+    if (annotations.count == 0)
+        return nil;
+
+    NSArray *visibleAnnotations = self.mapView.visibleAnnotations;
+
+    if (visibleAnnotations.count == annotations.count)
+        return nil;
+
+    NSMutableArray *invisibleAnnotations = [annotations mutableCopy];
+
+    if (visibleAnnotations.count > 0) {
+        [invisibleAnnotations removeObjectsInArray:visibleAnnotations];
+    }
+
+    // Now pick a random offscreen annotation.
+    uint32_t index = arc4random_uniform((uint32_t)invisibleAnnotations.count);
+    return invisibleAnnotations[index];
+}
+
+- (void)selectAnOffscreenAnnotation {
+    id<MGLAnnotation> annotation = [self randomOffscreenAnnotation];
+    [self.mapView selectAnnotation:annotation animated:NO];
```

Comment:
According to the [`-[MKMapView selectAnnotation:animated:]`](https://developer.apple.com/documentation/mapkit/mkmapview/1451950-selectannotation?language=objc) documentation, MapKit’s special case is because an offscreen annotation would have no backing view:

> If the specified annotation is not onscreen, and therefore does not have an associated annotation view, this method has no effect.

I think this SDK’s view-backed annotations will get selected just fine when off-screen, but I’m not sure what’ll happen once we implement #3249. This gives me pause: we should make sure #3249 is at least feasible for view-backed annotations before committing to the behavior here.

Reply:
I agree, I think it's worth waiting for #3249 before accepting these changes. I'll re-add the `do not merge` tag.